src/extensionsIntegrated/CSSColorPreview/main.js

```diff
@@ -65,27 +84,56 @@ define(function (require, exports, module) {
         const nLen = editor.lineCount();
         const aColors = [];
 
-        // match colors and push into an array
+        // Match colors and push into an array
         for (let i = 0; i < nLen; i++) {
             let lineText = editor.getLine(i);
+            let regx = /:[^;]*;/g;
+
+            let lineMatches = lineText.match(regx);
+            if (lineMatches) {
+                // Iterate through all matches to find valid colors
+                let validColors = [];
+
+                // We need to track the occurrence count for colors because in cases like this:
+                // /*color: linear-gradient(red, blue, green, yellow);*/ color: red;
+                // The color "red" appears twice: the first occurrence is inside a comment,
+                // but the second one is not. If we simply search for the index of "red"
+                // using `indexOf`, it will always return the first match, which belongs
+                // to the comment. This would incorrectly mark the second "red" (outside
+                // the comment) as invalid. By tracking occurrences, we ensure we check
+                // the correct instance of "red" and only ignore those inside comments.
+                let occurrenceCount = {}; // Track occurrences for each color
+
+                for (let lineMatch of lineMatches) {
```

Comment:
Instead of doing this, directly use the regex apis that returns index as well along with the result. This can be simplified.

Eg:
```js
below is pseudocode, just the general idea.
// Use matchAll to get both matches and their positions
let lineMatches = [...lineText.matchAll(regx)];
for (let lineMatch of lineMatches) {

let matches = [...lineMatches.matchAll(coolor_regex)]; // Convert iterator to an array

matches.forEach(match => {
    console.log(`Match: ${match[0]}, Position: ${lineMatch.index+match.index}`);
});
```
so now, you get the index directly and need not do anything else.